<a href="https://colab.research.google.com/github/mutisomartin/World-population/blob/main/RIB_(Lib).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd


In [ ]:
data = pd.read_csv("countries-table.csv")
data

place      pop1980       pop2000       pop2010       pop2023  \
0      356  696828385.0  1.059634e+09  1.240614e+09  1.428628e+09   
1      156  982372466.0  1.264099e+09  1.348191e+09  1.425671e+09   
2      840  223140018.0  2.823986e+08  3.111828e+08  3.399966e+08   
3      360  148177096.0  2.140724e+08  2.440162e+08  2.775341e+08   
4      586   80624057.0  1.543699e+08  1.944545e+08  2.404857e+08   
..     ...          ...           ...           ...           ...   
229    500      11452.0  5.138000e+03  4.938000e+03  4.386000e+03   
230    238       2240.0  3.080000e+03  3.187000e+03  3.791000e+03   
231    570       3637.0  2.074000e+03  1.812000e+03  1.935000e+03   
232    772       1647.0  1.666000e+03  1.367000e+03  1.893000e+03   
233    336        733.0  6.510000e+02  5.960000e+02  5.180000e+02   

          pop2024       pop2030       pop2050           country        area  \
0    1.441720e+09  1.514994e+09  1.670491e+09             India  3287590.00   
1    1.425179e+09  1.415606e+09  1.312636e+09             China  9706961.00   
2    3.418144e+08  3.521623e+08  3.753920e+08     United States  9372610.00   
3    2.797980e+08  2.921501e+08  3.172252e+08         Indonesia  1904569.00   
4    2.452098e+08  2.740298e+08  3.678085e+08          Pakistan   881912.00   
..            ...           ...           ...               ...         ...   
229  4.372000e+03  4.301000e+03  3.781000e+03        Montserrat      102.00   
230  3.803000e+03  3.869000e+03  3.779000e+03  Falkland Islands    12173.00   
231  1.935000e+03  1.948000e+03  2.096000e+03              Niue      261.00   
232  1.915000e+03  2.046000e+03  2.430000e+03           Tokelau       12.00   
233  5.260000e+02  5.610000e+02  7.310000e+02      Vatican City        0.44   

     landAreaKm cca2 cca3  unMember  netChange  growthRate  worldPercentage  \
0    2973190.00   IN  IND      True     0.4114      0.0092           0.1801   
1    9424702.90   CN  CHN      True    -0.0199     -0.0003           0.1780   
2    9147420.00   US  USA      True     0.0571      0.0053           0.0427   
3    1877519.00   ID  IDN      True     0.0707      0.0082           0.0350   
4     770880.00   PK  PAK      True     0.1499      0.0196           0.0306   
..          ...  ...  ...       ...        ...         ...              ...   
229      102.00   MS  MSR     False        NaN     -0.0032              NaN   
230    12173.00   FK  FLK     False        NaN      0.0032              NaN   
231      261.00   NU  NIU     False     0.0000      0.0000              NaN   
232       10.00   TK  TKL     False        NaN      0.0116              NaN   
233        0.44   VA  VAT      True        NaN      0.0154              NaN   

       density  densityMi  rank  
0     484.9067  1255.9084     1  
1     151.2174   391.6530     2  
2      37.3673    96.7813     3  
3     149.0254   385.9758     4  
4     318.0908   823.8551     5  
..         ...        ...   ...  
229    42.8627   111.0145   230  
230     0.3124     0.8091   231  
231     7.4138    19.2017   232  
232   191.5000   495.9850   233  
233  1195.4545  3096.2273   234  

[234 rows x 20 columns]

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 234 entries, 0 to 233
Data columns (total 20 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   place            234 non-null    int64  
 1   pop1980          234 non-null    float64
 2   pop2000          234 non-null    float64
 3   pop2010          234 non-null    float64
 4   pop2023          234 non-null    float64
 5   pop2024          234 non-null    float64
 6   pop2030          234 non-null    float64
 7   pop2050          234 non-null    float64
 8   country          234 non-null    object 
 9   area             234 non-null    float64
 10  landAreaKm       234 non-null    float64
 11  cca2             233 non-null    object 
 12  cca3             234 non-null    object 
 13  unMember         234 non-null    bool   
 14  netChange        224 non-null    float64
 15  growthRate       234 non-null    float64
 16  worldPercentage  228 non-null    float64
 17  density         

In [ ]:
drop_columns = ["place","cca2", "cca3", "unMember","densityMi"]
world_pop = data.drop(columns=drop_columns)
world_pop.head()

pop1980       pop2000       pop2010       pop2023       pop2024  \
0  696828385.0  1.059634e+09  1.240614e+09  1.428628e+09  1.441720e+09   
1  982372466.0  1.264099e+09  1.348191e+09  1.425671e+09  1.425179e+09   
2  223140018.0  2.823986e+08  3.111828e+08  3.399966e+08  3.418144e+08   
3  148177096.0  2.140724e+08  2.440162e+08  2.775341e+08  2.797980e+08   
4   80624057.0  1.543699e+08  1.944545e+08  2.404857e+08  2.452098e+08   

        pop2030       pop2050        country       area  landAreaKm  \
0  1.514994e+09  1.670491e+09          India  3287590.0   2973190.0   
1  1.415606e+09  1.312636e+09          China  9706961.0   9424702.9   
2  3.521623e+08  3.753920e+08  United States  9372610.0   9147420.0   
3  2.921501e+08  3.172252e+08      Indonesia  1904569.0   1877519.0   
4  2.740298e+08  3.678085e+08       Pakistan   881912.0    770880.0   

   netChange  growthRate  worldPercentage   density  rank  
0     0.4114      0.0092           0.1801  484.9067     1  
1    -0.0199     -0.0003           0.1780  151.2174     2  
2     0.0571      0.0053           0.0427   37.3673     3  
3     0.0707      0.0082           0.0350  149.0254     4  
4     0.1499      0.0196           0.0306  318.0908     5

In [ ]:
world_pop.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 234 entries, 0 to 233
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   pop1980          234 non-null    float64
 1   pop2000          234 non-null    float64
 2   pop2010          234 non-null    float64
 3   pop2023          234 non-null    float64
 4   pop2024          234 non-null    float64
 5   pop2030          234 non-null    float64
 6   pop2050          234 non-null    float64
 7   country          234 non-null    object 
 8   area             234 non-null    float64
 9   landAreaKm       234 non-null    float64
 10  netChange        224 non-null    float64
 11  growthRate       234 non-null    float64
 12  worldPercentage  228 non-null    float64
 13  density          234 non-null    float64
 14  rank             234 non-null    int64  
dtypes: float64(13), int64(1), object(1)
memory usage: 27.5+ KB


In [ ]:
world_pop.isna().sum()

pop1980             0
pop2000             0
pop2010             0
pop2023             0
pop2024             0
pop2030             0
pop2050             0
country             0
area                0
landAreaKm          0
netChange          10
growthRate          0
worldPercentage     6
density             0
rank                0
dtype: int64

In [ ]:
world_pop.isnull().any()

pop1980            False
pop2000            False
pop2010            False
pop2023            False
pop2024            False
pop2030            False
pop2050            False
country            False
area               False
landAreaKm         False
netChange           True
growthRate         False
worldPercentage     True
density            False
rank               False
dtype: bool

In [ ]:
world_pop.notnull().any()

pop1980            True
pop2000            True
pop2010            True
pop2023            True
pop2024            True
pop2030            True
pop2050            True
country            True
area               True
landAreaKm         True
netChange          True
growthRate         True
worldPercentage    True
density            True
rank               True
dtype: bool

In [ ]:
world_pop.notna().sum()

pop1980            234
pop2000            234
pop2010            234
pop2023            234
pop2024            234
pop2030            234
pop2050            234
country            234
area               234
landAreaKm         234
netChange          224
growthRate         234
worldPercentage    228
density            234
rank               234
dtype: int64

In [ ]:
world_pop.fillna(0, inplace= True)

In [ ]:
world_pop["worldPercentage"].fillna(0,inplace = True)

In [ ]:
world_pop["worldPercentage"].fillna(0,inplace = True)
world_pop["netChange"].fillna(0,inplace = True)
world_pop.isna().sum()

pop1980            0
pop2000            0
pop2010            0
pop2023            0
pop2024            0
pop2030            0
pop2050            0
country            0
area               0
landAreaKm         0
netChange          0
growthRate         0
worldPercentage    0
density            0
rank               0
dtype: int64

In [ ]:
world_pop.duplicated().sum()

0

**Observation**

-The data has 234 and 20 columns
-Some columns have missing data ('cca2' and 'world percentage', 'netchange')

Dropping some unnecessary columns

In [ ]:
world_pop.columns

Index(['pop1980', 'pop2000', 'pop2010', 'pop2023', 'pop2024', 'pop2030',
       'pop2050', 'country', 'area', 'landAreaKm', 'netChange', 'growthRate',
       'worldPercentage', 'density', 'rank'],
      dtype='object')

In [ ]:
len(world_pop.columns)

15

In [ ]:
world_pop.shape # (rows and columns)

(234, 15)

**Observations**

The dataset has 15 columns and 234 rows

In [ ]:
world_pop.insa().sum()

AttributeError: 'DataFrame' object has no attribute 'insa'